pymc/tests/test_distributions.py

```diff
@@ -952,6 +953,24 @@ def test_hierarchical_obs_logp():
     assert not any(isinstance(o, RandomVariable) for o in ops)
 
 
+@pytest.fixture(scope="module")
+def _compile_stickbreakingweights_logpdf():
+    _value = at.vector()
+    _alpha = at.scalar()
+    _k = at.iscalar()
+    _logp = logp(StickBreakingWeights.dist(_alpha, _k), _value)
+    return compile_pymc([_value, _alpha, _k], _logp)
+
+
+def _stickbreakingweights_logpdf(value, alpha, k, _compile_stickbreakingweights_logpdf):
+    return _compile_stickbreakingweights_logpdf(value, alpha, k)
+
+
+stickbreakingweights_logpdf = np.vectorize(
+    _stickbreakingweights_logpdf, signature="(n),(),(),()->()"
+)
```

Comment:
I am not sure if this is the one that should have the `pytest.fixture`, it seems like it should be a function that returns the vectorized `stickbreakingweights_logpdf`. The idea being that we only compile the Aesara function if the test that requires it, is run. Maybe it works like this already, if `np.vectorize` is lazy?

Otherwise my gut feeling is that it should be something like:

```suggestion
@pytest.fixture(scope="module")
def stickbreakingweights_logpdf():
   
    _value = at.vector()
    _alpha = at.scalar()
    _k = at.iscalar()
    _logp = logp(StickBreakingWeights.dist(_alpha, _k), _value)
    core_fn = compile_pymc([_value, _alpha, _k], _logp)
    
    return np.vectorize(_stickbreakingweights_logpdf, signature="(n),(),()->()")
```

Also it seems like you had too many parameters in the signature of np.vectorize? I removed one of the parenthesis.